### ISSUE TITLE
Feature request: autocomplete addresses.

### ISSUE THREAD

**Issue by eroncanc** [RefugeRestrooms/refugerestrooms#375]
Hi all,
I found about RR in a NB support fb group. Great work!

Would love to contribute. Is an autocomplete for addresses on app and website desired? Ive used the google maps autocomplete api before and it was pretty nice.

I saw talk about move to OSM, is that happening anytime soon?

**Comment by DeeDeeG** [comment]
> Is an autocomplete for addresses on app and website desired? Ive used the google maps autocomplete api before and it was pretty nice.

That sounds cool. I think as long as it doesn't cause any issues with our API rate limit (or increase fees) that would be a great addition.

Actually, I think we are already supposed to have auto-complete, at least on the splash page. Maybe it's not working because of some HTTPS issues we are having as of late? (#368)

> I saw talk about move to OSM, is that happening anytime soon?

OpenStreetMaps will happen whenever anyone researches how, and proves it can be done! (i.e. implement all the same features we have currently). I personally believe it can be done. If you wanted to look into that, I'd say go for it. I would be interested in making that happen as well and could be of some assistance if you like.

So the only thing holding that up is the R&D time it would take to do it, and we would also want to be sure the implementation is high-quality when it's done.

**Comment by eroncanc** [comment]
thanks for the reply! 
sorry for the late response, i was away but will begin to look into this and will use this issue to implement.

i believe the autocomplete endpoint uses separate rate limit, but will double check.

**Comment by DeeDeeG** [comment]
​Hi again eroncanc,

As I mentioned in my reply above, I thought we might already have auto-complete, but I did agree it wasn't working properly. I can confirm now: We do have it, it just doesn't load right away. It is blocked because of a thing happening with Piwik. If we were able to fix the Piwik issue, we would get autocomplete back.

If you want to know the full details, here is what I found going on with auto-complete at the moment:

We have a script from [Piwik](https://piwik.org/) which is intended to let us know some stats about what users/how many users visit Refuge Restrooms. (Piwik is like open-source Google Analytics.) Our copy is hosted on a server at enspiral.info (as discussed in #200). Our connection to the piwik.js file on that server must be bad, (Or perhaps the file is gone), because we attempt to download the file from there, but the connection times out.

The request for Piwik.js blocks any other requests for javascript files (only for requests that would begin after the piwik request starts) until it is done. 

(So, if we wait for our request for piwik.js to time out, the request for the Google Maps autocomplete API goes through, and auto-complete turns on. This could take a long time. On my computer and using Firefox, I see a page load time of a few seconds for the main page, but 3.12 minutes until the Maps API fully loads. On the same computer, but using Chromium, I get the main content in a few seconds, then the Maps api after about 49 seconds... and Chromium takes 2.2 minutes the second time.)

Regards,

DeeDeeG
